src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/ServiceAccountProvider.scala

```diff
@@ -0,0 +1,53 @@
+package org.broadinstitute.dsde.workbench.leonardo.model
+
+import com.typesafe.config.Config
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.model.{UserInfo, WorkbenchEmail}
+
+import scala.concurrent.Future
+
+/**
+  * Provides service accounts needed by Leo.
+  *
+  * @param config any necessary configuration information.
+  */
+abstract class ServiceAccountProvider(config: Config) {
```

Comment:
Yeah we're inconsistent in naming. :( I'm not a huge fan of `LeoServiceAccountProvider` though because it's misleading. It's providing service accounts, not Leo service accounts. `LeoAuthProvider` is a better because it's providing authorization specific to Leo operations.